lib/urlwatch/jobs.py

```diff
@@ -198,6 +199,11 @@ def retrieve(self, job_state):
         if result != 0:
             raise ShellError(result)
 
+        # If the first filter is a bytes filter, return content in bytes instead of in unicode
+        # as that's what's required by the library used by that filter
+        if self.filter and (FilterBase(self, self.filter) or FilterBase(self, self.filter[0])):
```

Comment:
```python
if self.filter and (Filterbase.is_bytes_filter(self.filter) or FilterBase.is_bytes_filter(self.filter[0]):
```

Reply:
Yeah, my internal test needs upgrading; thanks for the hint.

The correct code is actually `if self.filter and (FilterBase.is_bytes_filter(self.filter)
                            or any(FilterBase.is_bytes_filter(filter) for filter in self.filter[0])):`, although it's still technically incorrect because I should be checking the first key only, not all (using `any()`)

Reply:
Now only checking first key.

I can't think of anything else.  Thanks for your guidance and patience.